integrated_channels/sap_success_factors/exporters/learner_data.py

```diff
@@ -68,3 +69,53 @@ def get_learner_data_records(self, enterprise_enrollment, completed_date=None, g
                 'No learner data was sent for user [%s] because an SAP SuccessFactors user ID could not be found.',
                 enterprise_enrollment.enterprise_customer_user.username
             )
+
+
+class SapSuccessFactorsLearnerManger:
+    """
+    Class to manage SAPSF learners data and their relation with enterprise.
+    """
+
+    def __init__(self, enterprise_configuration, client=None):
+        """
+        Use the ``SAPSuccessFactorsAPIClient`` for content metadata transmission to SAPSF.
+
+        Arguments:
+            enterprise_configuration (required): SAPSF configuration connecting an enterprise to an integrated channel.
+            client: The REST API client that will fetch data from integrated channel.
+        """
+        self.enterprise_configuration = enterprise_configuration
+        self.client = client(enterprise_configuration) if client else None
+
+    def unlink_learners(self, **kwargs):
+        """
+        Iterate over each learner and unlink inactive SAP channel learners.
+
+        This method iterates over each enterprise learner and unlink learner
+        from the enterprise if the learner is marked inactive in the related
+        integrated channel.
+        """
+        enterprise_learners = EnterpriseCourseEnrollment.objects.select_related(
+            'enterprise_customer_user'
+        ).filter(
+            enterprise_customer_user__enterprise_customer=self.enterprise_configuration.enterprise_customer,
+            enterprise_customer_user__active=True,
+        ).order_by('enterprise_customer_user').values_list(
+            'enterprise_customer_user__user_id', flat=True
+        ).distinct()
+
+        for learner_id in enterprise_learners:
+            enterprise_learner = EnterpriseCustomerUser.objects.get(user_id=learner_id)
+
+            # TODO: Find if user is inactive on SAPSF
```

Comment:
Is it in process yet ?

Reply:
yes

Reply:
I have now implemented this funtionality